src/test/java/org/jenkinsci/plugins/workflow/cps/SnippetizerTester.java

```diff
@@ -103,6 +114,48 @@ protected Object invokeStep(StepDescriptor d, Object args) {
         r.assertEqualDataBoundBeans(step, actual);
     }
 
+    /**
+     * Recurses through the model of a {@link Describable} class's {@link DescribableModel} and its parameters to verify that doc generation will work.
+     *
+     * @param describableClass
+     *     A {@link Class} implementing {@link Describable}
+     * @throws Exception
+     *     If any errors are encountered other than {@link NoStaplerConstructorException}, which is ignored for now.
+     */
+    @SuppressWarnings("unchecked")
+    public static void assertDocGeneration(Class<? extends Describable> describableClass) throws Exception {
+        DescribableModel<?> model = new DescribableModel(describableClass);
+
+        assertNotNull(model);
+
+        recurseOnModel(model);
+
+    }
+
+    private static void recurseOnTypes(ParameterType type) throws Exception {
+        // For the moment, only care about types with @DataBoundConstructors.
+        if (type instanceof ErrorType && !(((ErrorType)type).getError() instanceof NoStaplerConstructorException)) {
```

Comment:
:ant: this seems fragile, because you will not know about `NoStaplerConstructorException`s which are genuine problems.

I would suggest adding a `Class<? extends Describable>...` parameter listing nested structs in the test classpath which you _know_ are messed up and you are OK with it for the moment.